metabox/metabox/core/machine.py

```diff
@@ -165,19 +166,25 @@ def get_late_setup(self):
         """
         return []
 
-    def start_remote(self, host, launcher, interactive=False, timeout=0):
+    def start_remote(self, host, launcher, cmd="remote", interactive=False, timeout=0):
+        """
+        Start remote
+        Note: This method will inject host and launcher at the end of cmd, if this is
+              undesirable, use run_cmd
+        """
+        if not cmd:
```

Comment:
The whole function signature is changed, and now it accepts a "cmd" , even the default is supplied, so I really don't understand why we have this check.

Looking at invocations of this function below AFAICT it's never called with anything other than cmd="", so why is the default value there in the first place.

Reply:
The default value documents what the default behaviour is, the if is there to "enforce' it because else one would have to check in calling functions what the value of cmd is and if it is "" then change it (this because Start is the same action for remote and non remote)